airflow/providers/amazon/aws/sensors/cloud_formation.py

```diff
@@ -0,0 +1,162 @@
+                 aws_conn_id='aws_default',
+                 poke_interval=60 * 3,
+                 *args,
+                 **kwargs):
+        super().__init__(poke_interval=poke_interval, *args, **kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.stack_name = stack_name
+        self.complete_status = complete_status
+        self.in_progress_status = in_progress_status
+        self.hook = None
+
+    def poke(self, context):
+        """
+        Checks for existence of the stack in AWS CloudFormation.
+        """
+        cloudformation = self.get_hook().get_conn()
+
+        self.log.info('Poking for stack %s', self.stack_name)
+
+        try:
+            stacks = cloudformation.describe_stacks(StackName=self.stack_name)['Stacks']
+            stack_status = stacks[0]['StackStatus']
+            if stack_status == self.complete_status:
+                return True
+            elif stack_status == self.in_progress_status:
+                return False
+            else:
+                raise ValueError(f'Stack {self.stack_name} in bad state: {stack_status}')
+        except ClientError as e:
+            if 'does not exist' in str(e):
+                if not self.allow_non_existing_stack_status():
+                    raise ValueError(f'Stack {self.stack_name} does not exist')
+                else:
+                    return True
+            else:
+                raise e
+
+    def get_hook(self):
+        """
+        Gets the AwsGlueCatalogHook
+        """
+        if not self.hook:
+            self.hook = AWSCloudFormationHook(aws_conn_id=self.aws_conn_id)
+
+        return self.hook
+
+    def allow_non_existing_stack_status(self):
+        """
+        Boolean value whether or not sensor should allow non existing stack responses.
+        """
+        return False
+
+
+class CloudFormationCreateStackSensor(BaseCloudFormationSensor):
+    """
+    Waits for a stack to be created successfully on AWS CloudFormation.
+
+    :param stack_name: The name of the stack to wait for (templated)
+    :type stack_name: str
+    :param aws_conn_id: ID of the Airflow connection where credentials and extra configuration are
+        stored
+    :type aws_conn_id: str
+    :param poke_interval: Time in seconds that the job should wait between each try
+    :type poke_interval: int
+    """
+
+    template_fields = ['stack_name']
+    ui_color = '#C5CAE9'
+
+    @apply_defaults
+    def __init__(self,
+                 stack_name,
+                 aws_conn_id='aws_default',
+                 poke_interval=60 * 3,
+                 *args,
+                 **kwargs):
+        super().__init__(stack_name=stack_name,
+                         complete_status='CREATE_COMPLETE',
+                         in_progress_status='CREATE_IN_PROGRESS',
+                         aws_conn_id=aws_conn_id,
+                         poke_interval=poke_interval,
+                         *args,
+                         **kwargs)
```

Comment:
My first question is "where is poke?", then I have to realize that there is a `BaseCloudFormationSensor` with the implementation.  I don't think it's the clearest way to write operators. 

What I would suggest is to implement in the hook a method `check_id_stack_exists` or `get_stack` and re-use it the poke in both sensors. Same can be done with `create_stack`, it should be hook method reused in operator.

Reply:
Sure, will move to the hook and get rid of the superclass.